### PR TITLE
Fix EZP-25339: Default value length should be between min and max string length

### PR DESCRIPTION
**Pull request by crevillo**
Fixes https://jira.ez.no/browse/EZP-25339

Requires https://github.com/ezsystems/repository-forms/pull/71 to work. 

ping @bdunogier @andrerom

### CODE REVIEW

**Comment by crevillo** [comment]
ping @gyle too :)

File: eZ/Publish/Core/FieldType/Validator/StringLengthValidator.php [hunk 1]
```diff
@@ -63,6 +64,30 @@ public function validateConstraints($constraints)
...
                     break;
+                case 'defaultValue':
+                    if ($value instanceof TextLineValue) {
+                        $value = $value->text;
+                    }
+                    $minStringLength = $constraints['minStringLength'];
+                    $maxStringLength = $constraints['maxStringLength'];
+                    if ($minStringLength !== false && is_integer($minStringLength) && (strlen($value) < $minStringLength)) {
+                        $validationErrors[] = new ValidationError(
+                            "Validator parameter '%parameter%' length can't be shorter than min string length value",
```
Comment (glye) [inline eZ/Publish/Core/FieldType/Validator/StringLengthValidator.php]: Nitpick: "min" isn't really a word, I believe we use "minimum" elsewhere. Same with "max" below.

**Comment by glye** [comment]
+1 aside from nitpick. Btw I prefer to delete @ezrobot comments after I've fixed it's complaints, to declutter the PR.

**Comment by crevillo** [comment]
@glye, can i delete those @ezrobot comments?

File: eZ/Publish/Core/FieldType/Validator/StringLengthValidator.php (hunk 1 above)
Comment (crevillo) [inline eZ/Publish/Core/FieldType/Validator/StringLengthValidator.php]: Nitpicks just fixed.

**Comment by bdunogier** [comment]
I suggest you reorganize the method a bit.

If you first test max & min length, you know if they're valid or not, and can skip `$maxStringLength !== false && is_integer($maxStringLength)` when testing the default value. The loop isn't really useful here, imho.

**Comment by crevillo** [comment]
yah... i don't understand that for / case thing too but didn't want to change much what was already done... 

So, are you suggesting to get ride of that loop?

**Comment by crevillo** [comment]
i guess loop avoids the need to check for ```$contraints['maxStringLength']``` existence

**Comment by crevillo** [comment]
taken the approach suggested by @bdunogier . Removed loop. Still there's the need to check $constraints keys to be sure key is one of the allowed ones. That's the reason for the first loop.

**Comment by glye** [comment]
> @glye, can i delete those @ezrobot comments?

Oh, maybe you can't. I can, got rid of the old ones :)

**Comment by crevillo** [comment]
StringLengtValidator test was failing cause it expected two messages when min and max string lenght have no correct values (one for each). In my code i was doing ```elseif``` all the one, so only one validation message was thrown for that test. let's see if travis agrees now.

**Comment by crevillo** [comment]
still failing one of the tests, but i don't see how the fail is related to this pr... 
ping for re-review @glye @bdunogier

**Comment by glye** [comment]
I have restarted the failing test. It was strange looking, like it might be a timing problem. If it fails the same way again we need to find out why. After all it's not impossible that a change in a fieldtype leads to problems in a content test.

**Comment by crevillo** [comment]
yep. let's see what it says now...

**Comment by glye** [comment]
Looks good, I'm still +1

**Comment by crevillo** [comment]
side question: about restarting failing tests, do we (pull requesters) have permissions to do that? or that is something only ez crew members can do?

**Comment by glye** [comment]
@crevillo I'm not sure. If you do, you'll find a recycle icon on the top right of the page of each build job.

**Comment by crevillo** [comment]
ok. i'll take a look next time. thanks @glye

**Comment by andrerom** [comment]
> @crevillo I'm not sure. If you do, you'll find a recycle icon on the top right of the page of each build job.

@glye this is only available to maintainers of a repo, so he won't have access to this

@crevillo there are several ways, rebase or ammend and force push is the cleanest.

**Comment by crevillo** [comment]
thanks @andrerom .

**Comment by crevillo** [comment]
gentle ping @bdunogier

**Comment by crevillo** [comment]
closed. not the way to go. may need a rethinking of the whole process. see https://github.com/ezsystems/repository-forms/pull/71

**Pull request closed without merging**